Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -1034,4 +1093,37 @@ namespace Terrain
         }
     }
 
+    AZ::Data::Instance<AZ::RPI::Buffer> TerrainMeshManager::CreateMeshBufferInstance(uint32_t elementSize, uint32_t elementCount, const void* initialData, const char* name)
+    {
+        AZ::RPI::CommonBufferDescriptor desc;
+        desc.m_poolType = AZ::RPI::CommonBufferPoolType::StaticInputAssembly;
+        desc.m_elementSize = elementSize;
+        desc.m_byteCount = desc.m_elementSize * elementCount;
+        desc.m_bufferData = initialData;
+
+        if (name != nullptr)
+        {
+            desc.m_bufferName = name;
+        }
+
+        return AZ::RPI::BufferSystemInterface::Get()->CreateBufferFromCommonPool(desc);
+    }
+
+    AZ::Data::Instance<AZ::RPI::Buffer> TerrainMeshManager::CreateRayTracingMeshBufferInstance(AZ::RHI::Format elementFormat, uint32_t elementCount, const void* initialData, const char* name)
+    {
+        AZ::RPI::CommonBufferDescriptor desc;
+        desc.m_poolType = AZ::RPI::CommonBufferPoolType::DynamicInputAssembly;
+        desc.m_elementSize = AZ::RHI::GetFormatSize(elementFormat);
+        desc.m_byteCount = desc.m_elementSize * elementCount;
+        desc.m_bufferData = initialData;
+        desc.m_elementFormat = elementFormat;
+
+        if (name != nullptr)
+        {
+            desc.m_bufferName = name;
+        }
+
+        return AZ::RPI::BufferSystemInterface::Get()->CreateBufferFromCommonPool(desc);
+    }
+
```

Comment:
I originally started with that, but it gets a little hard to follow due to the way that the RPI::CommonBufferDescriptor leverages the Unknown format to make decisions about the size.

It's much more clear to have it separated, and I don't think maintenance will be a problem since it's just building a small descriptor.